### PR TITLE
Add support for an absolute limit on capped tx cost based on a multiple of per-commit budget

### PR DESCRIPTION
**Pull request by aschran**
## Test plan 

Added tests

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-core/src/authority/authority_per_epoch_store.rs [hunk 1]
```diff
@@ -2846,10 +2846,13 @@ impl AuthorityPerEpochStore {
         // We track transaction execution cost separately for regular transactions and transactions using randomness, since
         // they will be in different PendingCheckpoints.
         let tables = self.tables()?;
-        let per_commit_budget = self
+        let default_per_commit_budget = self
             .protocol_config()
             .max_accumulated_txn_cost_per_object_in_mysticeti_commit_as_option()
             .unwrap_or(0);
+        let gas_budget_based_txn_cost_absolute_cap_commit_multiple = self
```
Comment (mystenmark) [inline crates/sui-core/src/authority/authority_per_epoch_store.rs:2853]: `multiplier` or `factor` might be a better word here, as `multiple` makes me think it contains multiple caps

File: crates/sui-core/src/authority/authority_per_epoch_store.rs [hunk 2]
```diff
@@ -2858,28 +2861,36 @@ impl AuthorityPerEpochStore {
...
-                per_commit_budget,
+                default_per_commit_budget,
                 &sequenced_transactions,
             )?,
             self.protocol_config().per_object_congestion_control_mode(),
             self.protocol_config()
                 .max_accumulated_txn_cost_per_object_in_mysticeti_commit_as_option(),
             self.protocol_config()
                 .gas_budget_based_txn_cost_cap_factor_as_option(),
+            gas_budget_based_txn_cost_absolute_cap_commit_multiple,
```
[the remark this answers is not in the document]
Comment (aschran) [reply crates/sui-core/src/authority/authority_per_epoch_store.rs:2872]: I agree this is getting a lot. I thought about that, but it seems like a violation of encapsulation that would ultimately make things worse

Then I'd have to have a "is it for randomness or not" parameter that really has nothing to do with how the congestion control is configured and just indicates which settings to use (essentially doing the same as what's already here), and would need to generate fake ProtocolConfigs and give a fake argument for "is it for randomness" in all the unit tests which really should just be able to directly configure the behavior of the object under test

Let me at least try to see if I can simplify things here

File: crates/sui-core/src/authority/authority_per_epoch_store.rs (hunk 1 above)
Comment (aschran) [reply crates/sui-core/src/authority/authority_per_epoch_store.rs:2853]: I wanted to express that it indicates a number of commits' worth of budget, multiplier or factor seem more generic and don't express that as well IMO. How about `commit_count`, do you think that's any better or still confusing?

File: crates/sui-core/src/authority/authority_per_epoch_store.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (aschran) [reply crates/sui-core/src/authority/authority_per_epoch_store.rs]: I did pretty much what you said, it at least simplifies `process_consensus_transactions_and_commit_boundary`

**Review by halfprice — APPROVED** [review]
LGTM!

**Pull request merged**